### PR TITLE
✨Feature/plugin non breaking spaces

### PR DESCRIPTION
**Pull request by filip-dot-cube**
— Adds plugin for replacing single-letter characters with non-breaking spaces

— So far only for Czech language, however patterns list could be extend via plugin options

— Later on we could base general find and replace plugin on this one

### CODE REVIEW

**Comment by filip-dot-cube** [comment]
https://www.youtube.com/watch?v=ZurS1L3cV48

File: packages/plugin-non-breaking-spaces/src/utils.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { mapKeys, mapValues } from "lodash";
+
+import { CustomPatterns } from ".";
```
Comment (horaklukas) [inline packages/plugin-non-breaking-spaces/src/utils.ts:3]: Are you sure about this import from `'.'` ? 🙂

File: packages/plugin-non-breaking-spaces/src/__tests__/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
+import { Line } from "@lokse/core";
+
+import nonBreakingSpacesPlugin from "..";
+
+import Transformer from "../../../core/lib/transformer/json";
```
Comment (horaklukas) [inline packages/plugin-non-breaking-spaces/src/__tests__/index.test.ts:5]: Please don't break packages encapsulation 🙏 

```suggestion
import { transformersByFormat, OutputFormat } from "@lokse/core";

const Transformer = transformersByFormat[OutputFormat.JSON];
```

File: packages/plugin-non-breaking-spaces/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        const pattern = patterns[language.toLowerCase()];
+        const replacement = options.useNbsp ? "$1$2&nbsp;" : "$1$2\u00A0";
+
+        const replaced = line.value.replace(pattern, replacement);
+
+        line.setValue(replaced);
+      } else {
+        options.logger.warn(
+          `Pattern for current language ${language} was not found`
+        );
```
[the remark this answers is not in the document]
Comment (horaklukas) [reply packages/plugin-non-breaking-spaces/src/index.ts:48]: 2. Ok, I understand, lets keep it as it is, I didn't realize the gramatics differencies the feature is based on. 

We could maybe provide default definitions for some most used languages like EN/DE. Please create an enhancement ticket for it 🙏

**Review by horaklukas — CHANGES_REQUESTED** [review]
Minor changes to make, but otherwise really good job 👏

File: packages/plugin-non-breaking-spaces/src/index.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (filip-dot-cube) [reply packages/plugin-non-breaking-spaces/src/index.ts:48]: Alright, makes sense.

https://github.com/AckeeCZ/lokse/issues/57

**Comment by filip-dot-cube** [comment]
> Minor changes to make, but otherwise really good job 👏 after that you're good to merge it 😉

Thanks :)

**Pull request merged**